.circleci/config.yml

```diff
@@ -386,6 +395,86 @@ jobs:
             mv ~/js_types/* ~/dist/
             ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} -delete ${CIRCLE_TAG} ~/dist/
 
+  publish-cocoapod:
+    macos:
+      xcode: '10.1.0'
+    steps:
+      - run:
+          name: update podspec
+          working_directory: release
+          command: |
+            VERSION=$(echo $CIRCLE_SHA1 | cut -c -7)
+            if [ "${CIRCLE_TAG}" != "" ]; then
+                VERSION=${CIRCLE_TAG}
+                VERSION=$(echo $VERSION | cut -c 2-)
+            fi
+            sed -i.bak 's/<version>/${VERSION}/g' TextileCore.podspec
+            pod trunk push TextileCore.podspec --allow-warnings
+
+  publish-aar:
+    docker:
+      - image: circleci/android:api-28-ndk
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: publish artifacts
+          working_directory: release
+          command: |
+            VERSION=$(echo $CIRCLE_SHA1 | cut -c -7)
+            if [ "${CIRCLE_TAG}" != "" ]; then
+                VERSION=${CIRCLE_TAG}
+                VERSION=$(echo $VERSION | cut -c 2-)
+            fi
+            sed 's/<VERSION>/${VERSION}/g' mobile-template.pom > mobile-${VERSION}.pom
+            cp ~/mobile.aar mobile-${VERSION}.aar
+            curl -i -X PUT -u '${BINTRAY_USERNAME}:${BINTRAY_API_KEY}' -T mobile-${VERSION}.pom https://api.bintray.com/maven/textile/maven/mobile/io/textile/mobile/${VERSION}/mobile-${VERSION}.pom;publish=1
+            curl -i -X PUT -u '${BINTRAY_USERNAME}:${BINTRAY_API_KEY}' -T mobile-${VERSION}.aar https://api.bintray.com/maven/textile/maven/mobile/io/textile/mobile/${VERSION}/mobile-${VERSION}.aar;publish=1
+            curl -i -X POST -u '${BINTRAY_USERNAME}:${BINTRAY_API_KEY}' https://api.bintray.com/content/textile/maven/mobile/${VERSION}/publish
+
+  publish-pb:
+    docker:
+      - image: circleci/android:api-28-ndk
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: publish artifacts
+          working_directory: release/PBProject
+          command: |
+            VERSION=$(echo $CIRCLE_SHA1 | cut -c -7)
+            if [ "${CIRCLE_TAG}" != "" ]; then
+                VERSION=${CIRCLE_TAG}
+                VERSION=$(echo $VERSION | cut -c 2-)
+            fi
+            cp ~/protos/* pb/src/main/java/io/textile/pb/
+            sed -i.bak 's/<version>/${VERSION}/g' pb/build.gradle
+            ./gradlew pb:install
+            ./gradlew pb:bintrayUpload
```

Comment:
Needs to use a skeleton android project to actually build the pb arifact for upload. The building and upload are both handled through gradle plugin commands.